packages/react-native-bpk-component-flat-list/src/common-types.js

```diff
@@ -20,25 +20,29 @@
 
 import PropTypes from 'prop-types';
 import { type Element } from 'react';
-import { Image } from 'react-native';
+import { ViewPropTypes } from 'react-native';
+import { type ViewStyleProp } from 'react-native/Libraries/StyleSheet/StyleSheet';
```

Comment:
We should probably stop importing it like this and do this instead: https://github.com/flow-typed/flow-typed/issues/631#issuecomment-426737331

Reply:
I get this error:
```
Error ┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈
packages/react-native-bpk-component-boilerplate/src/BpkBoilerplate.js:24:24

Cannot get React.ElementProps because property ElementProps is missing in
object type [1].

     packages/react-native-bpk-component-boilerplate/src/BpkBoilerplate.js
      21│ import React from 'react';
      22│ import { StyleSheet, View, ViewPropTypes } from 'react-native';
      23│
      24│ type ViewProps = React.ElementProps<typeof View>;
      25│ type ViewStyleProp = $PropertyType<ViewProps, 'style'>;
      26│
      27│ const styles = StyleSheet.create({

     /private/tmp/flow/flowlib_2abc87ac/react.js
 [1] 271│   declare export default {|
     272│     +DOM: typeof DOM,
     273│     +PropTypes: typeof PropTypes,
     274│     +version: typeof version,
     275│     +checkPropTypes: typeof checkPropTypes,
     276│     +createClass: typeof createClass,
     277│     +createContext: typeof createContext,
     278│     +createElement: typeof createElement,
     279│     +cloneElement: typeof cloneElement,
     280│     +createFactory: typeof createFactory,
     281│     +createRef: typeof createRef,
     282│     +isValidElement: typeof isValidElement,
     283│     +Component: typeof Component,
     284│     +PureComponent: typeof PureComponent,
     285│     +Fragment: typeof Fragment,
     286│     +Children: typeof Children,
     287│   |};
```

Reply:
Ah I have to explicitly import `{ type ElementProps } from 'react'`.

Reply:
Addressed this now 👍

Reply:
Weird, it should be available in our version, see: https://github.com/facebook/flow/blob/v0.78/lib/react.js#L250

Reply:
^^ https://github.com/Skyscanner/backpack-react-native/pull/64#discussion_r260276119

Reply:
`import type { Element, ElementProps } from 'react';` works :)

Reply:
Ah cool, didn't see your comment before posting mine 😄